etlman/projects/tests/test_forms.py

```diff
@@ -16,19 +16,26 @@ def test_django_form_fields(self):
                 "step_order": sf.step_order,
             }
         )
-        assert (
-            form.is_valid()
-        ), form.errors  # is valid automatically calls valid_data, so data is valid
+        assert form.is_valid(), form.errors  # is valid automatically calls valid_data
         saved_obj = form.save()
         assert sf.name == saved_obj.name
         assert sf.script == saved_obj.script
         assert sf.step_order == saved_obj.step_order
         assert sf.pipeline == saved_obj.pipeline
 
     def test_django_form_not_valid_duplicate_step_order(self):
-        # Test to write:
-        # - Create a Step and save it to the DB
-        # - Try to create a step in the same pipeline with a duplicate step_order
-        # - Check that the error exists per:
         # https://adamj.eu/tech/2020/06/15/how-to-unit-test-a-django-form/#unit-tests
-        pass
+        # Create a Step and save it to the DB
+        sf = StepFactory(pipeline=PipelineFactory())
+        # Try to create a step in the same pipeline with a duplicate step_order
+        form = StepForm(
+            data={
+                "name": sf.name,
+                "script": sf.script,
+                "pipeline": sf.pipeline,
+                "step_order": sf.step_order,
+            }
+        )
+        assert form.errors["__all__"], [
```

Comment:
Not quite -- might need `==` here? and `, form.errors` at the end, so you can see what the errors are if they don't match what you expect.